### PR TITLE
Updated basic RAG usage for LlamaIndex

### PR DESCRIPTION
**Pull request by timothyckl**
Updated RAG with LlamaIndex section to align with LlamaIndex v0.10

### CODE REVIEW

**Comment by sophiamyang** [comment]
Thanks @timothyckl! There was a conflict. Could you update the PR to include the newly added changes to the docs? Thanks!

File: docs/guides/03-basic-RAG.md [hunk 1]
```diff
@@ -213,23 +213,24 @@ Visit our [community cookbook example](https://github.com/mistralai/cookbook/blo
...
-from llama_index.embeddings import MistralAIEmbedding
-from llama_index import ServiceContext
-from llama_index.query_engine import RetrieverQueryEngine
+import os
+from llama_index.core import Settings, SimpleDirectoryReader, VectorStoreIndex
+from llama_index.llms.mistralai import MistralAI
+from llama_index.embeddings.mistralai import MistralAIEmbedding
+
+# Set Mistral API key
+os.environ["MISTRAL_API_KEY"]= "..."
```
Comment (sophiamyang) [inline docs/guides/03-basic-RAG.md:222]: I think we can remove this line

**Comment by timothyckl** [comment]
Happy to help @sophiamyang 😄  

Should be resolved now!

**Comment by sophiamyang** [comment]
Awesome! Merged!

**Pull request merged**